scripts/upload-tsv-files.py

```diff
@@ -0,0 +1,93 @@
+import argparse
+import boto3
+from datetime import datetime
+from more_itertools import one
+import os
+import requests
+import re
+import unicodedata
+import sys
+
+
+class ProjectTSVUploader():
+    def __init__(self, deployment):
+        deployments = {
+            'dev':
+                {
+                 'service_url': 'https://service.dev.explore.data.humancellatlas.org'
+                },
+            'prod':
+                {
+                 'service_url': 'https://service.explore.data.humancellatlas.org'
+                }
+        }
+
+        self.s3 = boto3.resource('s3')
+        self.service_url = deployments[deployment]['service_url']
+        self.deployment = deployment
+
+    # Copied function from
+    # https://github.com/DataBiosphere/azul/blob/5523064bbe2ffa7e6003d9d26105106229f0f93d/scripts/
+    # count-bundles.py#L12-L18
+    def _get_project_name(self, document_id):
+        base_url = f'{self.service_url}/repository/projects/{document_id}'
+        response = requests.get(base_url)
+        response.raise_for_status()
+        project_list = response.json()['projects']
+        one(project_list)
+        return project_list[0]['projectShortname']
+
+    def upload_files_to_bucket(self, bucket_name, project_tsv_directory):
+        key_prefix = 'projects/'
+        for filename in os.listdir(project_tsv_directory):
+            filepath = os.path.join(project_tsv_directory, filename)
+
+            if os.path.isfile(filepath):
+                key = f'{key_prefix}{filename}'
+                obj = self.s3.Object(bucket_name, key)
+                project_uuid = os.path.splitext(filename)[0]
+                project_name = self._get_project_name(project_uuid)
+                file_name = unicodedata.normalize('NFKD', project_name)
+                file_name = re.sub('[^\w ,.@%&-_()\\[\]/{}]', '_', file_name).strip()
+                timestamp = datetime.now().strftime("%Y-%m-%dT%H%M%S")
```

Comment:
BM_PC__2019-04-09T164408 is cryptic for most users. This should be a user-friendly timestamp, not one that's easily parsed. Drop the seconds. Use space to separate the timestamp from the short name. So `BM_PC 2019-04-09 16:44` instead of `BM_PC__2019-04-09T164408`

Reply:
If I use colons in the filename, like in `BM_PC 2019-04-09 16:44`, AWS corrects the content-disposition header by replacing the colon with an underscore. So after uploading the file, the filename becomes `BM_PC 2019-04-09 16_44`. Will that still be an acceptable filename?